### PR TITLE
Change hint "enable_plan_cache(bool)" to "ignore_plan_cache()"

### PR DESCRIPTION
**Pull request by lawyerphx**
### What problem does this PR solve? 
* [tidb 15587](https://github.com/pingcap/tidb/issues/15587)

### What is changed and how it works?
* Change hint "enable_plan_cache(bool)" to "ignore_plan_cache()"

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported variable/fields change

Side effects

Related changes

 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by lawyerphx** [comment]
Relevant pingcap/tidb PR : https://github.com/pingcap/tidb/issues/15587

File: hintparser.y [hunk 1]
```diff
@@ -78,7 +78,7 @@ import (
 
 	/* TiDB hint names */
 	hintAggToCop              "AGG_TO_COP"
-	hintEnablePlanCache       "ENABLE_PLAN_CACHE"
+	hintDisablePlanCache      "DISABLE_PLAN_CACHE"
```
Comment (zz-jason) [inline hintparser.y]: How about `ignore_plan_cache` or `ignore_cached_plan`? `disable_plan_cache` sounds like a hint that can disable the plan cache, clear all the items in the cache.

@eurekaka How do you think?

Comment (eurekaka) [reply hintparser.y]: `ignore_plan_cache` sounds reasonable.

Comment (lawyerphx) [reply hintparser.y]: rename done

**Pull request merged**